ophyd/areadetector/base.py

```diff
@@ -19,6 +20,77 @@ def __init__(self, prefix, **kwargs):
         super().__init__(prefix + '_RBV', write_pv=prefix, **kwargs)
 
 
+class NDDerivedSignal(DerivedSignal):
+    """
+    DerivedSignal to shape a flattened array
+
+    The purpose of this class is to take a flattened array and shape in its'
+    proper form. The shape of the final array may be static in which case the
+    shape and number of dimensions can be set as static integers. Otherwise,
+    other signals from this classes parent can inform what the proper shape of
+    the array
+    """
+    def __init__(self, derived_from, shape, num_dimensions,
+                 parent=None, **kwargs):
+        super().__init__(derived_from, parent=parent, **kwargs)
+        # Assemble our shape of signals
+        self._shape = []
+        for dim in shape:
+            if isinstance(dim, str):
+                dim = getattr(parent, dim)
+            self._shape.append(dim)
+        self._shape = tuple(self._shape)
+        # Assemble ndims
+        if isinstance(num_dimensions, str):
+            num_dimensions = getattr(parent, num_dimensions)
+        self._num_dimensions = num_dimensions
+
+        # Ensure callbacks are fired when array is reshaped
+        for dim in self._shape + (self._num_dimensions, ):
+            if not isinstance(dim, (int, float)):
```

Comment:
```suggestion
            if not isinstance(dim, int):
```

Float is not a valid type to reshape with